core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/environment/EnvironmentNetworkValidator.java

```diff
@@ -0,0 +1,50 @@
+package com.sequenceiq.cloudbreak.controller.validation.environment;
+
+import static com.sequenceiq.cloudbreak.api.endpoint.v4.common.mappable.CloudPlatform.AWS;
+import static com.sequenceiq.cloudbreak.api.endpoint.v4.common.mappable.CloudPlatform.AZURE;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.mappable.CloudPlatform;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.environment.base.EnvironmentNetworkAzureV4Params;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.environment.requests.EnvironmentNetworkV4Request;
+import com.sequenceiq.cloudbreak.controller.validation.ValidationResult.ValidationResultBuilder;
+
+@Component
+public class EnvironmentNetworkValidator {
+
+    public void validateNetwork(EnvironmentNetworkV4Request networkV4Request, String cloudPlatform, ValidationResultBuilder resultBuilder) {
```

Comment:
I will ask for a bigger refactor here. Create a Map<String, NetworkValidator> field. Init it in the @PostConstruct method with NetworkValidator implementations. Each Provider shall have it's own validation. Later on when we will support new providers: GCP, OpenStack, WhateverCloud, the new implementations will be new classes. 

It's important to design the code so it will be open for extension but closed for modification.